src/provider/view/components/NotificationView/NotificationView.tsx

```diff
@@ -92,15 +95,19 @@ function getGroupTitle(notification: StoredNotification, groupingType: GroupingT
 /**
  * Group notifications together based on the given grouping method e.g. application/date.
  * @param notifications List of notifications to sort into groups.
+ * @param applications A map of application registry to resolve the notification title
  * @param groupMethod Grouping type to use for sorting.
  */
-function groupNotifications(notifications: StoredNotification[], groupMethod: GroupingType): Map<string, Group> {
+function groupNotifications(notifications: StoredNotification[], applications: ApplicationMap, groupMethod: GroupingType): Map<string, Group> {
```

Comment:
Consider changing this to return an array rather than a map. We use a map internally to make it easier to construct our groups, but the user of this function just wants a list of groups, and doesn't care about its map nature

Reply:
Cool. This line - `[...groups.values()].map((group: Group) => (` - can just be `groups..map((group: Group) => (` now